.github/workflows/publish.yml

```diff
@@ -60,7 +60,14 @@ jobs:
         run: yarn build:swagger-gen
         env:
           AZ_AR_PASSWORD: ${{ secrets.AZ_AR_PASSWORD }}
+      - name: Download dist folders
+        uses: actions/download-artifact@v3
+        if: github.event_name == 'pull_request'
```

Comment:
Actually the check is not correct, it should be according to the trigger.
Maybe something like `github.event.release != null` or equivalent to see if you are triggered by `release` event or not

Reply:
if the event_name is pull_request then it couldn't be a release event 
else event_name would have been release no ?